src/index.js

```diff
@@ -50,52 +51,53 @@ function createLoader() {
     // transpiler to detect an incorrect dependency.
-    const loaderModule = __webpack_require__(require.resolve('aurelia-loader-webpack'));
-    return Promise.resolve(new loaderModule.WebpackLoader());
+    const m = __webpack_require__(require.resolve('aurelia-loader-webpack'));
+    return Promise.resolve(new m.WebpackLoader());
   }
 
   // SystemJS Loader Support
-  if (host.System && typeof host.System.import === 'function') {
+  if (host.System && typeof host.System.config === 'function') {
     return host.System.normalize('aurelia-bootstrapper').then(bsn => {
       return host.System.normalize('aurelia-loader-default', bsn);
     }).then(loaderName => {
       return host.System.import(loaderName).then(m => new m.DefaultLoader());
     });
   }
 
-  // Node.js Support
-  if (typeof global !== 'undefined' && typeof global.require !== 'undefined') {
-    /* note: we use a scoped global.require() instead of simply require()
-     * so that Webpack does not automatically include this loader as a dependency,
-     * similarly to the non-global call to System.import() before
-     */
-    const loaderModule = global.require('aurelia-loader-nodejs');
-    return Promise.resolve(new loaderModule.NodeJsLoader());
-  }
-
   // AMD Module Loader Support
-  if (typeof host.require === 'function') {
+  if (typeof host.require === 'function' && typeof host.require.version === 'string') {
     return new Promise((resolve, reject) => host.require(['aurelia-loader-default'], m => resolve(new m.DefaultLoader()), reject));
   }
 
+  // Node.js and Electron Support
+  if (isNodeLike && typeof module !== 'undefined' && typeof module.require !== 'undefined') {
+    // note: we use a scoped module.require() instead of simply require()
+    // so that Webpack's parser does not automatically include this loader as a dependency,
+    // similarly to the non-global call to System.import() above
+    const m = module.require('aurelia-loader-nodejs');
+    return Promise.resolve(new m.NodeJsLoader());
+  }
+
   return Promise.reject('No PLATFORM.Loader is defined and there is neither a System API (ES6) or a Require API (AMD) globally available to load your app.');
 }
 
 function initializePal(loader) {
   let type;
 
-  if (typeof window !== 'undefined') {
+  const isElectronRenderer = isNodeLike && process.type === 'renderer';
+
+  if (isNodeLike && !isElectronRenderer) {
+    type = 'nodejs';
+  } else if (typeof window !== 'undefined') {
     type = 'browser';
   } else if (typeof self !== 'undefined') {
     type = 'worker';
-  } else if (typeof global !== 'undefined') {
-    type = 'nodejs';
   } else {
     throw new Error('Could not determine platform implementation to load.');
   }
 
   return loader.loadModule('aurelia-pal-' + type)
-    .then(palModule => palModule.initialize());
+    .then(palModule => type === 'nodejs' && !isInitialized && palModule.globalize() || palModule.initialize());
```

Comment:
@EisenbergEffect the reason for `type === 'nodejs'` is that under Electron it might be true, yet we don't want to override the globals there, only initialize. Also, under Electron the default PAL will be pal-browser, BUT we want to support scenarios where the user chooses to load the pal-nodejs, because both should work, since Electron has both Node and Browser environments. That's where checking for `palModule.globalize === 'function'` might not be good enough.